src/.clang-tidy

```diff
@@ -9,6 +9,7 @@ performance-faster-string-find,
 performance-for-range-copy,
 performance-inefficient-vector-operation,
 performance-move-const-arg,
+performance-no-automatic-move,
```

Comment:
Maybe split this out to make review/testing easier. It may be related to https://github.com/bitcoin/bitcoin/pull/25429 and https://github.com/bitcoin/bitcoin/pull/26308

Reply:
> Maybe split this out to make review/testing easier. It may be related to #25429 and #26308

Done in #26758.